src/lib.rs

```diff
@@ -190,9 +190,22 @@ fn replace_dir<P: AsRef<Path>>(old: P, new: P) -> Result<()> {
     let new = new.as_ref();
 
     if old.exists() && old.is_dir() {
+        // sync the staging directory to ensure our changes have been persisted
+        let old_as_file = File::open(old)
+            .chain_err(|| format!("failed to open old dir '{}' for syncing", old.display()))?;
+        old_as_file.sync_all()
+            .chain_err(|| format!("failed to sync old dir '{}'", old.display()))?;
+
         truncate_dir(new)?;
```

Comment:
If the system crashes in the middle of `truncate_dir`, then I believe it's possible that we've already deleted some of the contents of `old` but the dentry for `new` hasn't reached disk yet.  Possible solutions:

- Additionally sync the parent dir before the rename, or
- Rename `new` out of the way, rename `old` to `new`, sync the parent, then delete the old `new`.

The second approach may also reduce the window during which `new` is incomplete if the original `new` contains many files.